include/onnxruntime/core/session/onnxruntime_cxx_api.h

```diff
@@ -2154,6 +2154,13 @@ struct AllocatorImpl : Base<T> {
   MemoryAllocation GetAllocation(size_t size);
   void Free(void* p);
   ConstMemoryInfo GetInfo() const;
+
+  /** \brief Function that returns the statistics of the allocator.
+   *
+   * \param[out] stats: A pointer to a KeyValuePairs object that will be filled with the allocator statistics.
+   * \return A Status indicating success or failure.
+   */
+  Status GetStats(KeyValuePairs* stats) const;
```

Comment:
nit: The C++ API implementation hides the status and throws on error, so `KeyValuePairs GetStats() const` would be more typical.

See the other implementations where ThrowOnError is called.